drivers/spi/spi_sam.c

```diff
@@ -396,14 +396,28 @@ static int spi_sam_transceive(struct device *dev,
 	return err;
 }
 
+static int spi_sam0_transceive_sync(struct device *dev,
+				    const struct spi_config *config,
+				    const struct spi_buf_set *tx_bufs,
+				    const struct spi_buf_set *rx_bufs)
+{
+	struct spi_sam0_data *data = dev->driver_data;
+
+	spi_context_lock(&data->ctx, false, NULL);
+	return spi_sam0_transceive(dev, config, tx_bufs, rx_bufs);
+}
+
 #ifdef CONFIG_SPI_ASYNC
 static int spi_sam_transceive_async(struct device *dev,
 				     const struct spi_config *config,
 				     const struct spi_buf_set *tx_bufs,
 				     const struct spi_buf_set *rx_bufs,
 				     struct k_poll_signal *async)
 {
-	return -ENOTSUP;
+	struct spi_sam0_data *data = dev->driver_data;
+
+	spi_context_lock(&get_dev_data(dev)->ctx, true, async);
```

Comment:
You went to the trouble to get `*data` above, you should use it here. Where is `get_dev_data()` defined anyway? I only see it as `static inline` in other spi drivers, which leads me to believe that this part of the driver isn't being built in CI...

Reply:
oh yes missed that